CheatSheets/angular-cheatsheet.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Angular CheatSheet
+description: Angular cheatsheet contains basics of angular binding & angular CLI.
+created: 2022-10-03
+updated: 2022-10-03
+---
+
+# Angular Cheatsheet
+
+Angular is an application-design framework and development platform for creating efficient and sophisticated single-page apps.
+
+#### Angular Binding:
+
+ - One Way Binding:
+	```
+	<p>title</p>
+	//No changes were reflected back to the variable.
+	```
+- Two Way Binding
+	```
+	<input [(ngMo­del­)]=­"­student.F­ir­stN­ame­">
+	//Changes were reflected back to the variable
+	```
+- Property Binding
+	```
+	<img [src]=­"­student.profilePicUrl">
+	```
+- Attribute Binding
+	```
+	<button [attr.a­ri­a-l­abe­l]=­"­ok">­Ok<­/bu­tto­n>
+	```
+- Class Binding
+	```
+	<div [class.Focused]="isFocused">S­ele­cte­d</­div>
+	```
+- ngClass
+	```
+	<div [ngClass]="assignClasses()">
+	  <h1>{{student.FirstName}}</h1>
+	</div>
+	```
+- Style Binding
+	```
+	<p [style.co­lor­]="i­sSe­lected ? 'green' : 'red'">Option {{i}}</p>
+	```
+ - ngStyle
+	 ```
+	<div [ngStyle]="setStyles()">
+	  {{student.name}}
+	</div>
+	```
+
+- Component Binding
+	```
+	<student-details [student]="currStudent"></student-details>
+	```
+- Directive Binding
+	```
+	<div [ngClass] = "­{se­lected: isSele­cte­d}">­Student<­/di­v>
+	```
+- Event Binding
+	```
+	<button (click­)="test()">­Test</­but­ton>
+	```
+- $event
+	```
+	<input [value]="student.name"
+	 (input)="student.name=$event.target.value">
+	 ```
+	 
+#### Angular Lifecycle Hooks:
+- ngOnInit()
+	```
+	It get invoked when angular initialize component or directive.
+	```
+- ngOnChanges()
+	```
+	It get invoked when angular sets data bound input property i.e. @Input().
+	```
```

Comment:
```suggestion
## Angular Lifecycle Hooks:

| Commands | Description |
| -------------- | ------------- |
| ngOnInit() | It get invoked when angular initialize component or directive. |
| ngOnChanges() | It get invoked when angular sets data bound input property i.e. @Input(). |
```
## Angular Lifecycle Hooks:

| Commands | Description |
| -------------- | ------------- |
| ngOnInit() | It get invoked when angular initialize component or directive. |
| ngOnChanges() | It get invoked when angular sets data bound input property i.e. @Input(). |

Please, create a `tabular format` rather than a `code block`. It is more convenient to understand.